services/QuillLMS/app/controllers/cms/districts_controller.rb

```diff
@@ -0,0 +1,253 @@
+      'Total Invoice Amount' => @district.total_invoice
+    }
+    @school_data = school_query
+    @admins = DistrictsAdmins.includes(:user).where(district_id: params[:id].to_i).map do |admin|
+      {
+        name: admin.user.name,
+        email: admin.user.email,
+        district_id: admin.district_id,
+        user_id: admin.user_id
+      }
+    end
+  end
+
+  def new_admin
+    @district = District.find(params[:id])
+  end
+
+  def add_admin_by_email
+    begin
+      user = User.find_by(email: params[:email_address])
+      district = District.find(params[:id])
+      DistrictsAdmins.create(user_id: user.id, district_id: district.id)
+      flash[:success] = "Yay! It worked! 🎉"
+      redirect_to cms_district_path(params[:id])
+    rescue
+      flash[:error] = "It didn't work! 😭😭😭"
+      redirect_back(fallback_location: fallback_location)
+    end
+  end
+
+  def remove_admin
+    admin = DistrictsAdmins.find_by(user_id: params[:user_id], district_id: params[:district_id])
+    flash[:error] = 'Something went wrong.' unless admin.destroy
+    flash[:success] = 'Success! 🎉'
+    redirect_back(fallback_location: cms_district_path)
+  end
+
+  def edit
+    @district = District.find(params[:id])
+    @editable_attributes = editable_district_attributes
+  end
+
+  def update
+    if District.find(edit_or_add_district_params[:id]).update(edit_or_add_district_params)
+      redirect_to cms_district_path(edit_or_add_district_params[:id])
+    else
+      render :edit
+    end
+  end
+
+  private def set_district
+    @district = District.find params[:id]
+  end
+
+  private def text_search_inputs
+    @text_search_inputs = ['district_name', 'district_city', 'district_state', 'district_zip', 'district_name', 'nces_id']
+  end
+
+  private def all_search_inputs
+    @text_search_inputs.map(&:to_sym) + [:sort, :sort_direction, :page, :premium_status]
+  end
+
+  private def district_query_params
+    params.permit(default_params + all_search_inputs)
+  end
+
+  private def district_query(params)
+    RawSqlRunner.execute(
+      <<-SQL
+        SELECT
+          districts.name AS district_name,
+          districts.city AS district_city,
+          districts.state AS district_state,
+          districts.zipcode AS district_zip,
+          districts.total_students AS total_students,
+          districts.total_schools AS total_schools,
+          districts.nces_id AS nces_id,
+          districts.id AS id,
+          districts.phone AS phone
+        FROM districts
+        #{where_query_string_builder}
+        GROUP BY
+          districts.name,
+          districts.nces_id,
+          districts.phone,
+          districts.city,
+          districts.state,
+          districts.zipcode,
+          districts.total_students,
+          districts.total_schools,
+          districts.id
+        #{order_by_query_string}
+        #{pagination_query_string}
+      SQL
+    ).to_a
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
Let's avoid adding new methods that have high CyclomaticComplexity. Let's see if we can refactor this to something more simple.